conda/cli/conda_argparse.py

```diff
@@ -121,21 +122,14 @@ def __init__(self, *args, **kwargs):
             self.description += "\n\nOptions:\n"
 
         pm = context.get_plugin_manager()
-        self._subcommands = sorted(
-            (
-                subcommand
-                for subcommands in pm.hook.conda_subcommands()
-                for subcommand in subcommands
-            ),
-            key=lambda subcommand: subcommand.name,
-        )
+        self._subcommands = sorted(chain.from_iterable(pm.hook.conda_subcommands()))
 
         # Check for conflicts
         seen = set()
         conflicts = [
             subcommand
             for subcommand in self._subcommands
-            if subcommand.name in seen or seen.add(subcommand.name)
+            if subcommand in seen or seen.add(subcommand)
```

Comment:
I had to stare at this one for a bit until I understood what was going on with the short circuit eval.  I think the logic might be clearer (but perhaps uglier?) written out as a loop but this isn't the place to make that call

Reply:
I'm not sure if we need to store the whole subcommand hook instance in the set, what do we get from that?

Reply:
@pkmooreanaconda I agree this isn't the clearest trick in the book, I've struggled for years trying to make this trick more obvious.

I considered making a new helper function where it would be easier to explain what is happening, something like:

```python
def unique(*iterables):
    # unique list of items in order of occurrence
    ordered = []
    # for every item in our iterables, we check to see if it has already been seen,
    # if it has, its added to the list of duplicates, otherwise it's appended to ordered
    duplicates = [
        item
        for item in chain.from_iterable(iterables)
        # shortcircuit for whether to add to duplicates list or to ordered list
        if item in ordered or ordered.append(item)
    ]
    return ordered, duplicates
```

And then in `conda/cli/conda_argparse.py`:

```python
self._subcommands, conflicts = unique(*pm.hook.conda_subcommands())
```

@jezdez you're right we don't need the entire object, I'd originally thought to define `conda.plugins.CondaSubcommand.__str__` to avoid the extra step of extracting the name in the conflicts error

Reply:
I would tend to want to just put this inline...

I see toolz unique() doesn't collect duplicates.

The old dict key : true trick could be used to do an ordered set, then iterate over the keys to return a list.

-1 on "if item in list"

Reply:
```
nonduplicates = {}
duplicates = [ item for item in chain.from_iterable(iterables) if nonduplicates.setdefault(item, True) ]
return list(nonduplicates.keys()), duplicates
```